packages/editor/src/components/widgets/code-editor/useCodeEditor.ts

```diff
@@ -13,9 +13,29 @@ export const monacoAutoFocus = (editor: monaco.editor.IStandaloneCodeEditor) =>
 
 export type ModifyAction = (value: string) => string;
 
-export const useMonacoEditor = (options?: { modifyAction?: ModifyAction; scriptAreaDatatypeEditor?: boolean }) => {
+export const useMonacoEditor = (options?: {
+  modifyAction?: ModifyAction;
+  maximizedCodeEditor?: {
+    cursorPosition: { lineNumber: number; column: number } | null;
+    selectionRange: monaco.IRange | null;
+  };
+}) => {
   const [editor, setEditor] = useState<monaco.editor.IStandaloneCodeEditor>();
 
+  if (editor) {
+    if (options?.maximizedCodeEditor) {
+      if (options.maximizedCodeEditor.cursorPosition !== null) {
+        editor.setPosition(options.maximizedCodeEditor.cursorPosition);
+      }
+      if (options.maximizedCodeEditor.selectionRange !== null) {
+        editor.setSelection(options.maximizedCodeEditor.selectionRange);
+      }
+      setTimeout(() => {
+        editor.focus();
+      }, 500);
+    }
+  }
```

Comment:
I don't fully understand why this inside this hook? There is already a concept on run some functions against the monaco editor `onMountFuncs` where you can add an array of functions which are run when the editor is mounted...
Handle this focus in a timeout screems to me 'I will behave very strange or break in the future'. And by the way there is already a monacoAutoFocus function which you can simply add to the onMountFuncs...
I would expect this selection stuff to be also something like that.